test/e2e/utils.go

```diff
@@ -73,3 +79,65 @@ func getThanosRules(body []byte, expGroupName, expRuleName string) error {
 	}
 	return fmt.Errorf("'%s' alert not found in '%s' group", expRuleName, expGroupName)
 }
+
+func createSelfSignedMTLSArtifacts(s *v1.Secret) error {
```

Comment:
Just a small suggestion: if you don't need to share the root CA between the writer and receiver, you can also use [GenerateSelfSignedCertKey](https://pkg.go.dev/k8s.io/client-go/util/cert#GenerateSelfSignedCertKey) to simplify this function.

Reply:
If we only needed a serving certificate we would use the [Service CA operator](https://github.com/openshift/service-ca-operator), but for mTLS we need a server certificate and a client certificate signed be the same CA (or at least a mutually trusted CAs).

Reply:
> You can usually specify the CA used for validating the certificate from the other end.
Not sure what you mean by "the other end" here?

We need a client cert that the server can verify and a server cert that the client can verify. We solve this here by creating on CA (that signed both the server and client cert) and passing that CA to both parties.

Reply:
Sorry about the confusion, what I was trying to suggest is shortening the function like this:
```
func createSelfSignedMTLSArtifacts(s *v1.Secret) error {
	certBytesServer, keyBytesServer, _ := cert.GenerateSelfSignedCertKey("openshift-cluster-monitoring-test", nil, nil)
	certBytesClient, keyBytesClient, _ := cert.GenerateSelfSignedCertKey("openshift-cluster-monitoring-test", nil, nil)

	s.Data["server.crt"] = certBytesServer
	s.Data["server.key"] = keyBytesServer
	s.Data["client.crt"] = certBytesClient
	s.Data["client.key"] = keyBytesClient
	s.Data["ca.crt"] = append(certBytesClient, certBytesServer...)

	return nil
}
```
However, I now see that this will fail because the instrumented app uses the same file for the tls certificate and CA certificate :(.

So let's not waste too much time with this since it might end up being a rabbit hole. Maybe once we replace the instrumented app with Prometheus as a remote write target we can reconsider this again.

Reply:
Another hitch here is that `GenerateSelfSignedCertKey` creates a server cert specifically. I.e. this certificate will be missing `x509.ExtKeyUsageClientAuth` in the `KeyUsage` list.

:+1: for revisiting this at a later stage.